models/MARBL_column/model_mod.f90

```diff
@@ -0,0 +1,638 @@
+
+    else
+        ! case where the requested depth is above the center of some layer, and below
+        ! the center of another. We interpolate linearly between the nearest layers above
+        ! and below.
+
+        !print *, "----------------------------------------------------"
+        !print *, "interpolating between layers ", layer_index, " and ", (layer_index + 1)
+
+        istatus(ens_index) = 0
+
+        ! computing the depths at the centers of the nearest layers above and below
+        depth_above = layerdepth_center
+        depth_below = layerdepth_top - layer_thicknesses(layer_index, ens_index) &
+                                        - .5*layer_thicknesses(layer_index + 1, ens_index)
+       
+        ! extracting the quantity values at the layers above and below
+        qty_index   = get_dart_vector_index(1, 1, layer_index, state_dom_id, qty_id)
+        state_slice = get_state(qty_index, state_handle)
+        val_above   = state_slice(ens_index)
+
+        qty_index   = get_dart_vector_index(1, 1, layer_index + 1, state_dom_id, qty_id)
+        state_slice = get_state(qty_index, state_handle)
+        val_below   = state_slice(ens_index)
+
+        !print *, "corresponding to the values: ", val_above, " and ", val_below
+        !print *, "----------------------------------------------------"
+
+        ! linear interpolation
+        expected_obs(ens_index) = val_above + (requested_depth - depth_above) * (val_below - val_above) / (depth_below - depth_above)
+
+        !print *, "final value = ",expected_obs(ens_index)
+    end if
+end do
+
+end subroutine model_interpolate
+
+
+
+!------------------------------------------------------------------
+! Returns the smallest increment in time that the model is capable 
+! of advancing the state in a given implementation, or the shortest
+! time you want the model to advance between assimilations.
+
+function shortest_time_between_assimilations()
+
+type(time_type) :: shortest_time_between_assimilations
+
+if ( .not. module_initialized ) call static_init_model
+
+shortest_time_between_assimilations = assimilation_time_step
+
+end function shortest_time_between_assimilations
+
+
+!------------------------------------------------------------------
+! Given an integer index into the state vector, returns the
+! associated location and optionally the physical quantity.
+
+subroutine get_state_meta_data(index_in, location, qty)
+
+integer(i8),         intent(in)  :: index_in
+type(location_type), intent(out) :: location
+integer,             intent(out), optional :: qty
+
+integer :: lon_index, lat_index, level, local_qty
+
+if ( .not. module_initialized ) call static_init_model
+
+call get_model_variable_indices(index_in, lon_index, lat_index, level, kind_index=local_qty)
+
+! Make sure the longitude is properly assigned
+geolon = station_location(1)
+geolat = station_location(2)
+
+if (geolon < 0.0_r8) geolon = 360.0 + geolon
```

Comment:
The station_location is constant, so you only have to this  code once in static_init_model.
The model is tiny, so performance wise this does not make a difference, but it might be clearer to have static things in static_model_init.

Reply:
good suggestion!